src/entities/subscribers/free-post-comment.subscriber.ts

```diff
@@ -0,0 +1,56 @@
+import { HistoryAction } from '@src/constants/enum';
+import { FreePostComment } from '@src/entities/FreePostComment';
+import { FreePostCommentHistory } from '@src/entities/FreePostCommentHistory';
+import {
+  EntitySubscriberInterface,
+  EventSubscriber,
+  InsertEvent,
+  UpdateEvent,
+} from 'typeorm';
+
+@EventSubscriber()
+export class FreePostCommentSubscriber
+  implements EntitySubscriberInterface<FreePostComment>
+{
+  listenTo() {
+    return FreePostComment;
+  }
+
+  async afterInsert(event: InsertEvent<FreePostComment>): Promise<void> {
+    await this.createHistory(event, HistoryAction.Insert);
+  }
+
+  async afterUpdate(event: UpdateEvent<FreePostComment>): Promise<void> {
+    let action: HistoryAction;
+
+    if (event.entity.deletedAt) {
+      action = HistoryAction.Delete;
+    } else {
+      action = HistoryAction.Update;
+    }
+
+    await this.createHistory(event, action);
+  }
+
+  private async createHistory(
+    event: InsertEvent<FreePostComment> | UpdateEvent<FreePostComment>,
+    action: HistoryAction,
+  ) {
+    const historyRepository = event.connection.getRepository(
+      FreePostCommentHistory,
+    );
+
+    const newHistory = historyRepository.create({
+      ...event.entity,
+      action,
+      freePostCommentId: event.entity.id,
+      id: undefined,
+      createdAt: undefined,
+    });
+
+    await historyRepository.save(newHistory, {
+      reload: false,
+      listeners: false,
+    });
```

Comment:
```suggestion
    await event.manager.getRepository(FreePostHistory).save(
      {
        ...event.entity,
        action,
        freePostId: event.entity.id,
        id: undefined,
        createdAt: undefined,
      },
      { reload: false, listeners: false },
    );
```
이렇게 하면 트랜잭션도 똑같이 묶이고 좀 더 깔끔해 보이는데 어떤가요?

Reply:
repository의 create method를 사용하면 entity class instance를 얻을 수 있어서 저렇게 짯는데 class instance가 필요한지 고민좀 해볼게요

Reply:
제안해주신 코드가 보기에는 더 깔끔하지만
event.entity로 들어오는 값이 신뢰할 수 없는 값이 들어올 가능성이 있어서 repo.create 메서드로 한번 가공하는게 좋을듯합니다.

Reply:
음 확실히 entity의 타입이 단순히 ObjectLiteral 타입이기 때문에 그 방법이 더 좋을 것 같네요